miden-lib/tests/test_miden_wallet.rs

```diff
@@ -0,0 +1,312 @@
+pub mod common;
+
+use assembly::{
+    ast::{ModuleAst, ProgramAst},
+    Assembler,
+};
+use common::{data::prepare_word, NodeIndex};
+use crypto::{Felt, StarkField, Word, ONE, ZERO};
+use miden_lib::{MidenLib, SatKernel};
+use miden_stdlib::StdLibrary;
+
+use miden_objects::{
+    assets::{Asset, FungibleAsset},
+    builder::DEFAULT_ACCOUNT_CODE,
+    mock::{
+        mock_account_storage, mock_inputs_with_existing, AssetPreservationStatus, MockAccountType,
+        ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN, ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_ON_CHAIN,
+        ACCOUNT_ID_SENDER,
+    },
+    notes::{Note, NoteOrigin, NoteScript},
+    Account, AccountCode, AccountId, AccountVault, BlockHeader, ChainMmr,
+};
+
+use miden_tx::{data::DataStore, TransactionExecutor};
+
+#[derive(Clone)]
+pub enum DataStoreError {
+    AccountNotFound(AccountId),
+    NoteNotFound(u32, NodeIndex),
+}
+
+#[derive(Clone)]
+
+pub struct MockDataStore {
+    pub account: Account,
+    pub block_header: BlockHeader,
+    pub block_chain: ChainMmr,
+    pub notes: Vec<Note>,
+}
+
+impl MockDataStore {
+    pub fn with_existing(account: Option<Account>, consumed_notes: Option<Vec<Note>>) -> Self {
+        let (account, block_header, block_chain, consumed_notes) = mock_inputs_with_existing(
+            MockAccountType::StandardExisting,
+            AssetPreservationStatus::Preserved,
+            account,
+            consumed_notes,
+        );
+        Self {
+            account,
+            block_header,
+            block_chain,
+            notes: consumed_notes,
+        }
+    }
+}
+
+impl DataStore for MockDataStore {
+    fn get_transaction_data(
+        &self,
+        account_id: AccountId,
+        block_num: u32,
+        notes: &[NoteOrigin],
+    ) -> Result<(Account, BlockHeader, ChainMmr, Vec<Note>), miden_tx::DataStoreError> {
+        assert_eq!(account_id, self.account.id());
+        assert_eq!(block_num as u64, self.block_header.block_num().as_int());
+        assert_eq!(notes.len(), self.notes.len());
+        let origins = self
+            .notes
+            .iter()
+            .map(|note| note.proof().as_ref().unwrap().origin())
+            .collect::<Vec<_>>();
+        notes.iter().all(|note| origins.contains(&note));
+        Ok((
+            self.account.clone(),
+            self.block_header.clone(),
+            self.block_chain.clone(),
+            self.notes.clone(),
+        ))
+    }
+
+    fn get_account_code(
+        &self,
+        account_id: AccountId,
+    ) -> Result<ModuleAst, miden_tx::DataStoreError> {
+        assert_eq!(account_id, self.account.id());
+        Ok(self.account.code().module().clone())
+    }
+}
+
+#[test]
+// Testing the basic Miden wallet - receiving an asset
+fn test_receive_asset_via_wallet() {
+    // Create assets
```

Comment:
I'm hoping most of the setup code in this test and in the other test can be eliminated with the new mock data framework. Ideally, it'll be some as simple as:
```Rust
let sender_account = mock_chain.get_account(0);
let recipient_account = mock_chain.get_account(1);
let note = mock_chain.get_note(0);
```

Reply:
Jup, I am hoping for that, too. And then we could do 

```
sender_account.code = format!("being call.foo end");

sender_account.transact([note], tx_script);

assert!(sender_account.vault.matic == 1);
```